fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/runners/FutureManagerRunner.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * FutureManagerRunner.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.runners;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabaseRunner;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContextConfig;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A simple runner that extends {@link TransactionalRunner} by completing futures once the runner closes.
+ * Futures created externally can be registered with this runner. This runner can also create new futures (and register them).
+ * Once registered, the incomplete futures will be completed exceptionally (with a
+ * {@link com.apple.foundationdb.record.provider.foundationdb.FDBDatabaseRunner.RunnerClosed} exception) once the runner closes.
+ * It is the responsbility of the users of the runner to ensure futures that it creates are registered.
+ * <p>
+ * Normally, only the top level (root) of the future chain needs registration. Completing this future
+ * will cause the completion to trickle down to the dependent futures.
+ */
+@API(API.Status.INTERNAL)
+public class FutureManagerRunner extends TransactionalRunner {
```

Comment:
I _feel_ like it would be better to make this a separate class, and not extend `TransactionalRunner`. That gives two benefits:
1. You would be able to test this without having to worry about the database and contexts.
2. It would be usable in situations where it's not tied to a `TransactionalRunner`

I'm not sure those justify changing this as part of this PR.

Reply:
Good point. Tests, in particular, are now much easier to implement. Done.
Care would have to be taken when orchestrating this with other runners (e.g. order of closing, etc). If this becomes a concern, we can put together a combined class for this + runner.